services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/HankeServiceITests.kt

```diff
@@ -1201,6 +1204,90 @@ class HankeServiceITests(
         }
     }
 
+    @Nested
+    inner class GenerateHankeWithJohtoselvityshakemus {
+        private val hakemusNimi = "Johtoselvitys for a private property"
+
+        @Test
+        fun `saves hanke based on request`() {
+            val request = CreateHankeRequest(hakemusNimi, DEFAULT_HANKE_PERUSTAJA)
+
+            hankeService.generateHankeWithJohtoselvityshakemus(request, setUpProfiiliMocks())
+
+            assertThat(hankeRepository.findAll()).single().all {
+                prop(HankeEntity::generated).isTrue()
+                prop(HankeEntity::status).isEqualTo(HankeStatus.DRAFT)
+                prop(HankeEntity::nimi).isEqualTo(hakemusNimi)
+                prop(HankeEntity::createdByUserId).isEqualTo(USERNAME)
+                prop(HankeEntity::createdAt).isRecentUTC()
+            }
+        }
+
+        @Test
+        fun `saves hakemus based on request`() {
+            val request = CreateHankeRequest(hakemusNimi, DEFAULT_HANKE_PERUSTAJA)
+
+            hankeService.generateHankeWithJohtoselvityshakemus(request, setUpProfiiliMocks())
+
+            assertThat(applicationRepository.findAll())
+                .single()
+                .prop(ApplicationEntity::applicationData)
+                .prop(ApplicationData::name)
+                .isEqualTo(hakemusNimi)
+        }
+
+        @Test
+        fun `returns the saved hakemus`() {
+            val request = CreateHankeRequest(hakemusNimi, DEFAULT_HANKE_PERUSTAJA)
+
+            val hakemus =
+                hankeService.generateHankeWithJohtoselvityshakemus(request, setUpProfiiliMocks())
+
+            assertThat(hakemus).all {
+                prop(Hakemus::applicationData).prop(HakemusData::name).isEqualTo(hakemusNimi)
+            }
+        }
+
+        @Test
+        fun `generates hankekayttaja for founder`() {
+            val request = CreateHankeRequest(hakemusNimi, DEFAULT_HANKE_PERUSTAJA)
+
+            val hakemus =
+                hankeService.generateHankeWithJohtoselvityshakemus(request, setUpProfiiliMocks())
+
+            val hanke = hankeRepository.findByHankeTunnus(hakemus.hankeTunnus)!!
+            val kayttajat = hankekayttajaRepository.findByHankeId(hanke.id)
+            assertThat(kayttajat.first()).all {
```

Comment:
Does using `first()` instead of `single()` implicate that there are more than one hankekayttaja created? How do we then know for sure that the founder is the first in result list?

Reply:
Yes, using `single()` would be better here. I copied this from the old test and didn't notice this.

I changed this to `single()`. Thank you for the recommendation.